demos/2d_compressible_ALA/2d_compressible_ALA.py

```diff
@@ -136,7 +136,9 @@
 # pass in different nullspace and transpose_nullspace objects to the stokes solver.
 
 # +
-energy_solver = EnergySolver(T, u, approximation, delta_t, ImplicitMidpoint, bcs=temp_bcs)
+energy_solver = EnergySolver(
```

Comment:
Is there a reason you changed the order from T,u, approximation -> approximation, T, u...? 

The PR would be neater if this would be the same as before and it wouldnt break existing code.

Reply:
I think that it is a discussion I wanted to raise. So happy to have devs' thoughts on it. I do not mind reverting changes. Also, I doubt any of my PRs will ever be neat.

Reply:
I agree that the parameter order should probably be kept the same.

Reply:
The council has spoken. Reverted.

Reply:
Looking at `update_approx` today, I remembered why I changed the order of arguments. Given that temperature becomes optional in `StokesSolver`, it now has a default value (`fd.Constant(0.0)`). `approximation`, on the other hand, is not optional and does not have a default value. Therefore, it must precede `T` in the initialisation call. So that left two options: `z, approximation, T` or `approximation, z, T`. I thought the second one made more sense, hence why my initial proposed changes here. Happy to hear your thoughts.

Reply:
This isn't `update_approx`, so I think it doesn't need to be making premature API changes. It's much easier to reason about the changes if the API stays the same. If it eventually becomes optional in `StokesSolver`, then it could make sense to make a more sweeping change to the order of arguments.